vcd/resource_vcd_vapp_vm.go

```diff
@@ -123,6 +137,31 @@ func vmSchemaFunc(vmType typeOfVm) map[string]*schema.Schema {
 			Description:  "The amount of RAM (in MB) to allocate to the VM",
 			ValidateFunc: validateMultipleOf4(),
 		},
+		"memory_reservation": &schema.Schema{
+			Type:        schema.TypeInt,
+			Optional:    true,
+			Computed:    true,
+			Description: "The amount of RAM (in MB) reservation on the underlying virtualization infrastructure",
+		},
+		"memory_priority": &schema.Schema{
+			Type:         schema.TypeString,
+			Optional:     true,
+			Computed:     true,
+			Description:  "Pre-determined relative priorities according to which the non-reserved portion of this resource is made available to the virtualized workload",
+			ValidateFunc: validation.StringInSlice([]string{"LOW", "NORMAL", "HIGH", "CUSTOM"}, false),
+		},
+		"memory_shares": &schema.Schema{
+			Type:        schema.TypeInt,
+			Optional:    true,
+			Computed:    true,
+			Description: "Custom priority for the resource. This is a read-only, unless the `memory_priority` is CUSTOM",
+		},
+		"memory_limit": &schema.Schema{
+			Type:        schema.TypeInt,
+			Optional:    true,
+			Computed:    true,
+			Description: "The limit for how much of memory can be consumed on the underlying virtualization infrastructure. This is only valid when the resource allocation is not unlimited.",
```

Comment:
This comes from documentation, As far as understand it means valid when VDC limit is -1.

Reply:
Putting some context with UI screenshot.


I think it is valid in UI, but not valid here because in UI you do one of the following:
*  either "tick" the "unlimited" checkbox (and then input field for limit becomes greyed out) . I think the `This is only valid when the resource allocation is not unlimited.` comes from here
* either put a value in the limit field

In our case - both of these settings are set using the same field - that is why we don't need it. `-1` value in the field means `unlimited` in our case.

<img width="560" alt="image" src="https://user-images.githubusercontent.com/15804230/152130253-4b7d1d72-6fef-49d7-bbe9-c303b2fcc8d9.png">
.

Reply:
we sync on zoom, updated